### PR TITLE
Concurrent traversal utilities 

### PR DESCRIPTION
**Pull request by roysc**
* <strike>stripped down Nodeiterator interface</strike> fulfills trie.NodeIterator interface
* prefix-bound iterator for traversing a specific range of keys
* <strike>visitor over subtries divided at a specific depth</strike> splits trie domain into (power of 2) covering bins

### CODE REVIEW

**Comment by i-norden** [comment]
Looking good! I still need to complete my review but I think this would be a good place for some unit tests, also we should go ahead and move the packages up to the root directory since there won't be a cmd/ directory in this repository (just a pure package like https://github.com/vulcanize/ipfs-ethdb)

**Comment by roysc** [comment]
Added some commits to correct issues found during statediff integration (https://github.com/vulcanize/eth-statediff-service/issues/5).

**Comment by i-norden** [comment]
Still need to move stuff up to the root path and include some unit tests for the iterator itself, that's not the priority but needs to be done before merging this PR.

**Review by i-norden — COMMENTED** [review]
Still need unit tests in this repository for iterator itself and to move the stuff in pkg/ up to the root path. I can't review this in depth today, but we should go ahead and cut a release off of this `dev` branch to use with the `eth-statediff-service`.

File: pkg/iterator/iterator.go
Comment (roysc) [reply pkg/iterator/iterator.go]: yup, debugging cruft

**Review by i-norden — APPROVED** [review]
Looks great! I can't find anything else to add, if you move the stuff in `iterator/` up to the root path I think we can go ahead and merge this. We can't consider the task complete until there are unit tests in this repo for the iterator itself, because we need to use the iterator in more than just the context of the `eth-statediff-service`, but let's not let that hold up this PR any longer.

**Pull request merged**
